### PR TITLE
File format checks

### PR DESCRIPTION
**Pull request by MancunianSam**

### CODE REVIEW

File: src/test/scala/uk.gov.nationalarchives.fileformat/ExternalServicesTest.scala [hunk 1]
```diff
@@ -0,0 +1,72 @@
+package uk.gov.nationalarchives.fileformat
+
+import java.io.File
```
Comment (TomJKing) [inline src/test/scala/uk.gov.nationalarchives.fileformat/ExternalServicesTest.scala:3]: `import java.io.File` appears to be unnecessary import

File: src/test/scala/uk.gov.nationalarchives.fileformat/FileUtilsTest.scala [hunk 2]
```diff
@@ -0,0 +1,181 @@
+package uk.gov.nationalarchives.fileformat
+
+import java.nio.file.{Path, Paths}
```
Comment (TomJKing) [inline src/test/scala/uk.gov.nationalarchives.fileformat/FileUtilsTest.scala:3]: `..., Paths}` appears to be an unnecessary import

File: src/test/scala/uk.gov.nationalarchives.fileformat/LambdaTest.scala [hunk 3]
```diff
@@ -0,0 +1,68 @@
+package uk.gov.nationalarchives.fileformat
+
+import java.util.UUID
+
+import graphql.codegen.types.FFIDMetadataInput
+import io.circe.parser.decode
+import org.scalatest.matchers.should.Matchers.{equal, _}
+import uk.gov.nationalarchives.fileformat.AWSUtils._
+
+import scala.util.Try
```
Comment (TomJKing) [inline src/test/scala/uk.gov.nationalarchives.fileformat/LambdaTest.scala:10]: `import scala.util.Try` appears to be unnecessary import

File: Jenkinsfile-deploy [hunk 4]
```diff
@@ -0,0 +1,7 @@
+@Library("tdr-jenkinslib@lambda-methods") _
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply Jenkinsfile-deploy:1]: I've changed this one and the other one I missed.

File: Jenkinsfile-deploy [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+
+deployToLambda(
+        stage: params.STAGE,
+        libraryName: "file-format",
+        version: params.TO_DEPLOY
+)
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply Jenkinsfile-deploy:7]: I've added a break in.

File: Jenkinsfile-test [hunk 6]
```diff
@@ -0,0 +1,6 @@
+@Library("tdr-jenkinslib@lambda-methods") _
+
+lambdaTests(
+        stage: params.STAGE,
+        libraryName: "file-format"
+)
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply Jenkinsfile-test:6]: I've added this one in.

File: Jenkinsfile-build [hunk 7]
```diff
@@ -0,0 +1,57 @@
+library("tdr-jenkinslib")
+
+def repo = "tdr-file-format"
+pipeline {
+    agent none
+
+    parameters {
+        choice(name: "STAGE", choices: ["intg", "staging", "prod"], description: "The stage you are building the front end for")
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply Jenkinsfile-build:8]: I've updated this.

File: src/main/resources/application.conf [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+auth {
+    client {
+        id = "set_me_client_id"
+        id = ${?CLIENT_ID}
+        secret = "set_me_secret"
+        secret = ${?CLIENT_SECRET}
+    }
+}
+
+command = "sf"
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply src/main/resources/application.conf:37]: :+1:

File: src/test/resources/application.conf [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+auth {
+    client {
+        id = "client_id"
+        secret = "secret"
+    }
+}
+
+command = "test.sh"
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply src/test/resources/application.conf:30]: :+1:

File: src/main/resources/application.conf [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+        input = "https://sqs.eu-west-2.amazonaws.com/set_me_account_number/tdr-file-format-intg"
+        input = ${?INPUT_QUEUE}
+        output = "https://sqs.eu-west-2.amazonaws.com/set_me_account_number/tdr-api-update-intg"
+        output = ${?OUTPUT_QUEUE}
+    }
+}
+
+efs {
+    root {
+        location = "/mnt/fileformat"
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply src/main/resources/application.conf:17]: No, it's not an environment thing, all lambda mount points have to begin with /mnt/ for AWS reasons.

File: src/main/scala/uk/gov/nationalarchives/fileformat/FileUtils.scala [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+
+import java.nio.file.Paths
+import java.util.UUID
+
+import com.amazonaws.services.lambda.runtime.events.models.s3.S3EventNotification.S3EventNotificationRecord
+import com.typesafe.config.ConfigFactory
+import graphql.codegen.GetOriginalPath.getOriginalPath.{Data, Variables, document}
+import software.amazon.awssdk.services.s3.S3Client
+import software.amazon.awssdk.services.s3.model.GetObjectRequest
+import uk.gov.nationalarchives.fileformat.SiegfriedResponse._
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/fileformat/FileUtils.scala:11]: Yeah, you're right, I've removed it.

File: src/main/scala/uk/gov/nationalarchives/fileformat/Lambda.scala [hunk 12]
```diff
@@ -0,0 +1,43 @@
+package uk.gov.nationalarchives.fileformat
+
+import com.amazonaws.services.lambda.runtime.Context
+import com.amazonaws.services.lambda.runtime.events.SQSEvent
+import com.typesafe.config.{Config, ConfigFactory}
+import software.amazon.awssdk.services.sqs.model.DeleteMessageResponse
+import uk.gov.nationalarchives.aws.utils.Clients.{s3, sqs}
+import uk.gov.nationalarchives.aws.utils.S3EventDecoder._
+import uk.gov.nationalarchives.aws.utils.SQSUtils
+import uk.gov.nationalarchives.tdr.keycloak.KeycloakUtils
```
[the remark this answers is not in the document]
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/fileformat/Lambda.scala:10]: It is, and the s3 import above is as well, I've taken that out.

File: src/test/scala/uk.gov.nationalarchives.fileformat/ExternalServicesTest.scala (hunk 1 above)
Comment (MancunianSam) [reply src/test/scala/uk.gov.nationalarchives.fileformat/ExternalServicesTest.scala:3]: It is, I've taken out the line breaks around it as well :smile:

File: src/test/scala/uk.gov.nationalarchives.fileformat/FileUtilsTest.scala (hunk 2 above)
Comment (MancunianSam) [reply src/test/scala/uk.gov.nationalarchives.fileformat/FileUtilsTest.scala:3]: Yep, I've taken it out.

File: src/test/scala/uk.gov.nationalarchives.fileformat/LambdaTest.scala (hunk 3 above)
Comment (MancunianSam) [reply src/test/scala/uk.gov.nationalarchives.fileformat/LambdaTest.scala:10]: I've removed this one as well.

**Pull request merged**